### ISSUE TITLE
"aws cloudformation (deploy|wait)": describe-stack polling gets AWS throttling API errors

### ISSUE THREAD

**Issue by powersjcb** [aws/aws-cli#2948]
The AWS API throttles requests after what appears to be a set number of API calls. We recently switched several workflows to heavily rely on `aws cloudformation (deploy|wait)`. These operations are consistently being throttled during peak usage from an excess number of `describe-stack` API requests.

As an alternative to AWS support limbo, waiting for a throttling increase, it would be nice to manually reduce throughput of waiters for tasks users know will take more time.

The failure:
`An error occurred (Throttling) when calling the DescribeStacks operation (reached max retries: 4): Rate exceeded`

Related ticket #1376

**Comment by joguSD** [comment]
This seems reasonable marking as a feature request.
@sanathkr

**Comment by sanathkr** [comment]
Yeah this is actually a good feature request. I will comment on the PR

**Comment by ASayre** [comment]
Good Morning!

We're closing this issue here on GitHub, as part of our migration to [UserVoice](https://aws.uservoice.com/forums/598381-aws-command-line-interface) for feature requests involving the AWS CLI.

This will let us get the most important features to you, by making it easier to search for and show support for the features you care the most about, without diluting the conversation with bug reports.

As a quick UserVoice primer (if not already familiar): after an idea is posted, people can vote on the ideas, and the product team will be responding directly to the most popular suggestions.

We’ve imported existing feature requests from GitHub - Search for this issue there!

And don't worry, this issue will still exist on GitHub for posterity's sake.  As it’s a text-only import of the original post into UserVoice, we’ll still be keeping in mind the comments and discussion that already exist here on the GitHub issue.

GitHub will remain the channel for reporting bugs.  

Once again, this issue can now be found by searching for the title on: https://aws.uservoice.com/forums/598381-aws-command-line-interface 

-The AWS SDKs & Tools Team

**Comment by sbe-arg** [comment]
IMO

aws cloudformation deploy | wait backend Describe commands should be by default unlimited and handling rate limit properly unless explicitly --set to stop at X count.

**Comment by sbe-arg** [comment]
cross reference https://github.com/aws/aws-cli/issues/3665#issuecomment-635902479

**Comment by tim-finnigan** [comment]
Checking in as this issue hasn't had any activity in a while. We have been advised to refer uses to the [SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) and the [sam deploy](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html) command going forward. Have you tried using the SAM CLI? 

I'll also link this AWS CLI documentation on configuring retries because I think it's relevant here and I didn't see it linked in this issue: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html
